crates/example-types/src/block_types.rs

```diff
@@ -29,7 +30,7 @@ impl TestTransaction {
     ///
     /// # Errors
     /// If the transaction length conversion fails.
-    pub fn encode(transactions: Vec<Self>) -> Result<Vec<u8>, BlockError> {
+    pub fn encode(transactions: &Vec<Self>) -> Result<Arc<Vec<u8>>, BlockError> {
```

Comment:
```suggestion
    pub fn encode(transactions: &[Self]) -> Result<Vec<u8>, BlockError> {
```
- `transactions` type should be either `&[Self]` or `Vec<Self>` but not `&Vec<Self>`
- No need to return `Arc<Vec<u8>>` because the returned `Vec` is newly created. The caller can wrap it in `Arc` if needed.

Reply:
Ok. Modified according to your suggestion so the caller creates the `Arc` (if needed).